From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Tue, 16 Aug 2016 11:08:43 +0200
Subject: [PATCH] {For LLVMLinux} compiler-clang.h: add unreachable()
 definition

This fixes objtool warnings such as:

   arch/x86/kernel/cpuid.o: warning: objtool: cpuid_smp_cpuid()+0x90:
   function has unreachable instruction

Without this, unreachable() is compiled as "do { } while (1)" (it is its
definition in compiler.h), which is correctly detected as unreachable
instructions by objtool.
---
 include/linux/compiler-clang.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/linux/compiler-clang.h b/include/linux/compiler-clang.h
index b1ce500fe8b3..0e119cda68fc 100644
--- a/include/linux/compiler-clang.h
+++ b/include/linux/compiler-clang.h
@@ -44,3 +44,6 @@
 #define __must_be_array(a) BUILD_BUG_ON_ZERO(__same_type((a), &(a)[0]))
 #define __assume_aligned(a, ...)	\
 	__attribute__((__assume_aligned__(a, ## __VA_ARGS__)))
+
+#define unreachable() \
+	do { annotate_unreachable(); __builtin_unreachable(); } while (0)
-- 
